react-app/src/account/Account.js

```diff
@@ -0,0 +1,86 @@
+import scrypt from 'scrypt-async';
+import nacl from 'tweetnacl';
+import sha256 from 'fast-sha256';
+import blake from 'blakejs';
+
+nacl.util = require('tweetnacl-util');
+
+class Account {
+  constructor(username, password) {
+    const hashedPassword = sha256(nacl.util.decodeUTF8(password));
+    const salt = sha256(nacl.util.decodeUTF8(username));
+    const options = {
+      N: 1 << 20,
+      r: 8,
+      p: 1,
+      dkLen: 32,
+      encoding: 'binary'
+    };
+
+    const callback = (derivedKey) => {
+      const keyPair = nacl.box.keyPair.fromSecretKey(derivedKey);
+
+      const systemTzOffsetInMinutes = -(new Date()).getTimezoneOffset();
```

Comment:
Timezone, offset and username setting shouldn't be part of this callback. This callback is called to notify us that the keys are ready and we should handle only that. The rest of the assignments can be done on the constructor itself.
This change will make more sense once we take the key creation out of the constructor since it's async.
We may want to consider extracting the timezone data calculation into a utility function, the logic itself can be tested separately and doesn't depend on the account. I'm guessing that this is a data that the server needs and we need to get it right, it may be useful to have it on its own to test/document extensively.

Reply:
I set this values inside the callback because I understood that scrypt() is actually synchronous, and I wanted to set all the values together when they are all ready, but it makes much more sense if they are outside of the callback, I agree.

Reply:
Done